### ISSUE TITLE
[Progress] Research use-cases and create `Loading` component

### ISSUE THREAD

**Issue by jjenzz** [radix-ui/primitives#1127]
I'm wondering if our `Progress` might currently be little too low-level (like `Announce`) as there are different a11y requirements depending on its use-case.

> If the `progressbar` is describing the loading progress of a particular region of a page, the author **SHOULD** use `aria-describedby` to point to the status, and set the `aria-busy` attribute to `true` on the region until it is finished loading.
> https://www.w3.org/TR/wai-aria-1.2/#progressbar

In this case, it seems we should provide a `Loading` component that uses `Progress` under the hood. 

Other use-cases might be:

- When application is making progress toward completing a requested action, e.g. Upload progress
- Steps in a form https://www.w3.org/WAI/tutorials/forms/multi-page/#using-the-html5-progress-element
- Others? Research needed

The above seem to be covered by our current `Progress` implementation so perhaps all we're missing is a `Loading` abstraction.

**Comment by luisorbaiceta** [comment]
I'd love to work on that Loading abstraction if you decide to go on with it!

**Comment by jjenzz** [comment]
I think this one will need quite a bit of research before the work starts as I'm not sure a single `Loading` component is all that's needed. For example, we might need an anatomy that follows something like:

```jsx
<ViewLoader>
  <ViewLoader.Spinner />
</ViewLoader>
```

The `ViewLoader` would be the region with the `aria-describedby` and `aria-busy` attributes.

There might be other issues/features that are more ready to start on tbh. We have been wanting to start on a `MenuBar` for some time if that's something you'd be interested in? https://github.com/radix-ui/primitives/issues/1133
